MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxNetworkRequests.java

```diff
@@ -253,9 +263,10 @@ public void post(URL url, JSONObject body, Hashtable<String, String> headers) {
   }
 
   @Override
-  public void postWithCompletion(String propertyKey, String body,
+  public void postWithCompletion(String domain, String propertyKey, String body,
       Hashtable<String, String> headers,
       INetworkRequest.IMuxNetworkRequestsCompletion callback) {
+    beaconDomain = domain;
```

Comment:
I think that given there is a member variable of beaconDomain now rather than modifying every call to postWithCompletion it would make sense to simply use the set beaconDomain and expect anyone that wants to change this to call setDomain in advance, while preserving the old method signature. Having beaconDomain quietly change as a consequence of this in addition to having a setter seems quite non-obvious.

Reply:
That is a bad idea, having beaconDomains changed in the middle of a session will result in broken stats on a backend. beaconDomains are actually set in MuxStats constructor as an CustomerOption parameter and it need to be same for entire MuxStats life cycle, in case the user want to send data to different domain it need to initialize new MuxStats object.

Reply:
So the domain passed in to postWithCompletion has to match the beaconDomain class variable or you could get broken stats? I think the confusion is why do we have both?

Reply:
We do not need both, we could pass beaconDomain to getAuthorety function as an argument, if you think that is a cleaner solution.

Reply:
Using the `getAuthority` method seems much cleaner. But I also think we should try to achieve this without breaking API compatibility.